pycernan/avro/client.py

```diff
@@ -1,47 +1,137 @@
 """
     Base Avro client from which all other clients derive.
 """
+import contextlib
 import socket
 
 import pycernan.avro.config
 
 from abc import ABCMeta, abstractmethod
+from queue import Queue, Empty
 
 from pycernan.avro.exceptions import EmptyBatchException
 from pycernan.avro.serde import serialize
 
 
+DefunctConnection = None
+
+
+class EmptyPoolException(Exception):
```

Comment:
Should this go in `pycernan.avro.exceptions`?

Reply:
Yes, it should.  I will move.

Reply:
Moved.